src/vs/workbench/contrib/terminal/electron-browser/terminalInstance.ts

```diff
@@ -739,14 +739,33 @@ export class TerminalInstance implements ITerminalInstance {
 		}
 		if (this._processManager) {
 			this._processManager.dispose(immediate);
+		} else {
+			// In cases where there is no associated process (for example executing an extension callback task)
+			// consumers still expect on onExit event to be fired. An example of this is terminating the extension callback
+			// task. There is no exit code at this point, so firing undefined is appropriate.
+			this._onExit.fire(undefined);
 		}
+
 		if (!this._isDisposed) {
 			this._isDisposed = true;
 			this._onDisposed.fire(this);
 		}
 		this._disposables = lifecycle.dispose(this._disposables);
 	}
 
+	public finishedWithRenderer(result?: number): void {
```

Comment:
Let's change this to `rendererExit(exitCode: number): void`, only a number should be passed through here.

Reply:
@alexr00 and @Tyriar - See my comment about (number | undefined) above. I will revert this if you feel strongly.